101-machine-learning-linkedserivce-create/azuredeploy.json

```diff
@@ -0,0 +1,125 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "amlWorkspaceName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the AML workspace"
+            }
+        },
+        "linkName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the LinkedService"
+            }
+        },
+        "synapseWorkspaceResourceId": {
+            "type": "string",
+            "metadata": {
+                "description": "ResourceId of the Synapse workspace which you want to link to the aml workspace."
+            }
+        },
+        "sparkPools": {
+            "type": "array",
+            "defaultValue": [
+                {
+                    "computeName": "compute1",
+                    "poolName": "sparkPool1"
+                },
+                {
+                    "computeName": "compute2",
+                    "poolName": "sparkPool2"
+                }
+            ],
+            "metadata": {
+                "description": "Spark pools you want to attach to the aml workspace. Pool name is the name of spark pool shown in Synapse workspace, while compute name is the alias for this pool after it is attached to aml workspace."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "allowedValues": [
+                "australiaeast",
+                "brazilsouth",
+                "canadacentral",
+                "centralus",
+                "eastasia",
+                "eastus",
+                "eastus2",
+                "francecentral",
+                "japaneast",
+                "koreacentral",
+                "northcentralus",
+                "northeurope",
+                "southeastasia",
+                "southcentralus",
+                "uksouth",
+                "westcentralus",
+                "westus",
+                "westus2",
+                "westeurope"
+            ],
+            "metadata": {
+                "description": "Location(aka region) of the LinkedService"
+            }
+        },
+        "subscriptionId": {
+            "type": "string",
+            "defaultValue": "[subscription().id]"
+        },
+        "resourceGroup": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().name]"
+        }
+    },
+    "functions": [],
+    "variables": {
+        "apiVersion": "2020-04-01-preview"
+    },
+    "resources": [
+        {
+            "name": "[concat(parameters('amlWorkspaceName'), '/', parameters('linkName'))]",
+            "type": "Microsoft.MachineLearningServices/workspaces/linkedServices",
+            "apiVersion": "[variables('apiVersion')]",

```

Comment:
this must be a literal apiVersion - fix and remove the variable

Reply:
will do. But may I ask why we must use a literal apiVersion rather than a variable here?